### ISSUE TITLE
[feature request] Rewrite "_vacuumIndex" in the way so it doesn't use a recursion technique

### ISSUE THREAD

**Issue by vladimiry** [localvoid/ndx#31]
The recursion technique is not very well scalable so the "Maximum call stack size exceeded"-like error occurs on some datasets.

**Comment by localvoid** [comment]
I agree that it should be fixed, I just haven't touched js for more than a year, so it is unlikely that I'll write code to fix this issue.
Another way to avoid this issue could be to use `termFilter` that trims suffix. Something like `(term) => term.substring(0, 24)`
